Core/Src/main.c

```diff
@@ -87,14 +103,47 @@ int main(void)
   /* Initialize all configured peripherals */
   MX_GPIO_Init();
   MX_USART2_UART_Init();
+  MX_SPI1_Init();
+  MX_TIM1_Init();
   /* USER CODE BEGIN 2 */
 
+
+
+  	  HAL_GPIO_WritePin(GPIOB, GPIO_PIN_8, GPIO_PIN_SET); //Initially CS high
+
+  	  TIM_HandleTypeDef htim1;
+
+  	  HAL_TIM_PWM_START(&htm1, TIM_CHANNEL_1);
+
+  	  uint16_t pwm_val = 0;
+
   /* USER CODE END 2 */
 
   /* Infinite loop */
   /* USER CODE BEGIN WHILE */
   while (1)
   {
+	  HAL_Delay(10);
+
+	  //ADC Communication Code
+	  uint8_t txData[3] = {FIRST_BYTE, SECOND_BYTE, THIRD_BYTE}; //does this need to be an array of pointers?
+	  // ie. uint8_t *txData[3] = {FIRST_BYTE, SECOND_BYTE, THIRD_BYTE};
+	  uint8_t rxData[3] = {0};
```

Comment:
This looks good to me but it can be defined outside the while look and also, if txData will never change, it should be const. And it does not need to be an arrray of pointers.